server-spi/src/main/java/org/keycloak/models/UserModel.java

```diff
@@ -129,9 +138,15 @@
 
     void removeRequiredAction(String action);
 
-    void addRequiredAction(RequiredAction action);
+    default void addRequiredAction(RequiredAction action) {
+        String actionName = action.name();
```

Comment:
There should be check for `action != null` (even though I appreciate it's not there in the original JPA variant)